### PR TITLE
Run with ssl enabled

### PR DESCRIPTION
**Pull request by aionick**
## Description

Please include a brief summary of the change that this pull request proposes. Include any relevant motivation and context. List any dependencies required for this change.

- Users now able to run aion using a self-signed ssl certificate. Additionally, I added some command line options to the aion.sh script to create the directory where the certificate will be placed and to create the certificate. Once merged, will need to update the wiki with a short description of how to use this functionality.

Fixes Issue #510.

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [x] New feature.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [x] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- All manual tests, verifying a number of scearios: able to access rpc-json end points over https and not http when option is enabled; multiple certs; bad password; missing cert; missing keystore dir, etc.
- To generate a self-signed cert you will need <a href="https://www.openssl.org/source/">OpenSSL</a> and at command `cd` into project root and do `./aion.sh -s store` to create the keystore directory and then `./aion.sh -s create` (there is a second command to customize host & ip as well) and follow the prompts to create the certificate.
- In IDEA you will need to add the sslKeystore directory to the modApiServer classpath.

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [x] Any dependent changes have been made.

### CODE REVIEW

File: modMcf/src/org/aion/mcf/config/CfgApiRpc.java [hunk 1]
```diff
@@ -50,6 +48,7 @@
         this.corsEnabled = false;
         this.maxthread = 1;
         this.filtersEnabled = true;
+        this.ssl = new CfgApiRpcSsl();
```
Comment (AionJayT) [inline modMcf/src/org/aion/mcf/config/CfgApiRpc.java]: suggest just use CfgSSL as the class name

File: modAionImpl/src/org/aion/zero/impl/cli/Cli.java [hunk 2]
```diff
@@ -113,6 +108,14 @@ public int call(final String[] args, final Cfg cfg) {
...
                     System.out.println("p2p: " + cfg.getNet().getP2p().getIp() + ":" + cfg.getNet().getP2p().getPort());
                     break;
+                case "-s":
+                    File store = new File(CfgApiRpcSsl.SSL_KEYSTORE_DIR);
+                    if (store.mkdir()) {
+                        System.out.println("\nCreated sslKeystore directory.");
+                    } else {
+                        System.out.println("\nThe sslKeystore directory already exists.");
+                    }
+                    break;
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/cli/Cli.java]: I think -s should generate the ssl file for the user also create the folder. not just create a folder.
Which means we might warp the openssl tool into binary release.

**Comment by aionick** [comment]
I spaced out the printHelp menu a bit because originally my flags had more arguments and I needed the space. I forgot to put it back to how it was before but don't want to pollute the PR with a bunch of tiny commits, so I'll wait for reviews before I change it so I can modify more than just that

**Comment by qoire** [comment]
Unavailable for review, will defer to @ali-sharif

**Comment by ali-sharif** [comment]
Reviewing now. Will get back with my review before tommorow.

**Comment by ali-sharif** [comment]
I'm seeing weird SSL exceptions non-deterministically from NanoHttpd when I hammer it with SSL requests. Let's Icebox this PR. I'm gonna just push this with the PR for https://github.com/aionnetwork/aion/issues/519 sometime next week.

**Comment by ali-sharif** [comment]
Iceboxed. SSL will be included in my httpserver-update PR.

**Pull request closed without merging**